aws/rust-runtime/aws-sigv4/src/event_stream.rs

```diff
@@ -102,7 +103,7 @@ pub fn sign_message<'a>(
 ) -> Result<SigningOutput<Message>, SigningError> {
     let message_payload = {
         let mut payload = Vec::new();
-        message.write_to(&mut payload).unwrap();
+        write_message_to(message, &mut payload).unwrap();
```

Comment:
nit: Just seeking further clarification: Had we introduced a `Writer` trait with a `write_to` method, and a parallel `Reader` trait with a `read_from` function in the `aws_smithy_eventstream::frame` module, could we have implemented these traits for both `Header` and `Message`? This way, had we been able to retain the familiar API calls like `message.write_to`, `header.write_to`, and `Message::from`?

Reply:
Yes. Using a verb-style name, if we had the `Write` trait and the `Read` trait, `Message`, `Header`, and `HeaderValue` can each implement their own `write_to` and `read_from`. However, defining the signature for `read_from` has a small twist because `read_header_from` also returns the number of bytes read, whereas `Message` and `HeaderValue` just return `Self`.
```
fn read_header_from<B: Buf>(mut buffer: B) -> Result<(Header, usize), Error>
```
Still, we could implement such traits, but that can be later given `aws-smithy-eventstream` is an unstable crate.

Reply:
Yes. Using a verb-style name, if we had the `Write` trait and the `Read` trait, `Message`, `Header`, and `HeaderValue` could each implement their own `write_to` and `read_from`. However, defining the signature for `read_from` has a small twist because `read_header_from` also returns the number of bytes read, whereas `Message` and `HeaderValue` just return `Self` (so we probably need a generic over a return type).
```
fn read_header_from<B: Buf>(mut buffer: B) -> Result<(Header, usize), Error>
```
Still, we could implement such traits, but that can be later given `aws-smithy-eventstream` is an unstable crate.